katana/katana.py

```diff
@@ -90,6 +93,207 @@ def readconfig():
+                            kw_list_1.append("{0}.{1}({2})".format(class_list, kw_list, arg_name))
+                            count += 1
+                    doc_string_value.append("The keyword {0} in Driver {1} has a defined "
+                                            "arguments {2} You must want to send other values"
+                                            " through data file".format(kw_list, driver, arg_name))
+                else:
+                    for argument in argument_list:
+                        argument_name = argument.get('name')
+                        if argument_name is not None:
+                            arg_value = argument.get('value')
+                            if arg_value is None or arg_value is False:
+                                arg_value = argument.text
+                            arg_list = argument_name + "=" + "\'" + arg_value + "\',"
+                            arg_name_list += arg_list
+                            count += 1
+                    if len_arg_list == count:
+                        kw_list_1.append("{0}.{1}({2})".format(class_list, kw_list, arg_name_list))
+                    doc_string_value.append("The keyword {0} in Driver {1} has a defined arguments "
+                                            "{2} You must want to send other values through data "
+                                            "file".format(kw_list, driver, arg_name_list))
+                    count = 0
+                    arg_name_list = ""
+        sum_val = 1
+        for _, new in enumerate(doc_string_value):
+            sample.append('\n' + "        " + str(sum_val) + ". " + new)
+            sample_string = "".join(sample)
+            sum_val += 1
+            if sum_val > len(subkw_list):
+                break
+        for line in io.open('kw_seq_template', 'r'):
+            with open(ActionFile, 'a+') as f2:
+                line = line.replace('$new_line', '\n')
+                line = line.replace('$wrapper_kw', WrapperName)
+                line = line.replace('$doc_string_values', sample_string)
+                if Description is not None:
+                    line = line.replace('$wdesc', Description)
+                else:
+                    line = line.replace('$wdesc', 'Description not provided by the user')
+                line = line.replace('$kw_list_1', str(kw_list_1))
+                f2.write(line)
+        with open(ActionFile, 'a+') as f2:
+            for _, vals in enumerate(import_action_list):
+                vals = vals.decode('utf-8')
+                f2.write('\n' + "        " + vals)
+            for _, val in enumerate(object_list_new):
+                val = val.decode('utf-8')
+                f2.write('\n' + "        " + val)
+            for line in io.open('kw_seq_temp', 'r'):
+                f2.write('\n' + "    " + line)
+        f2.close()
+    return checkval
+
+
+def getChildTextbyParentTag(datafile, pnode, cnode):
```

Comment:
To avoid excessive fileIO, please modify this helper function to take in xml etree instead of filename

Reply:
removed this method since I have not required it to get the values.